src/orbital-helpers/orbital_space.cc

```diff
@@ -299,8 +304,261 @@ void make_avas(psi::SharedWavefunction ref_wfn, psi::Options& options, psi::Shar
+				}
+			}
+		}
+
+		if (options.get_str("CUTOFF_BY") == "CUM_THRESHOLD") {
+			double tmp = 0.0;
+			double sum_lo = 0.0;
+			double sum_lv = 0.0;
+			for (int i = 0; i < nroccpi[0]; i++) {
+				sum_lo += lo->get(0, i);
+			}
+			for (int i = 0; i < nrvirpi[0]; i++) {
+				sum_lv += lv->get(0, i);
+			}
+
+			double cum_l_o = 0.0;
+			for (int i = 0; i < nroccpi[0]; i++) {
+				tmp += lo->get(0, i);
+				cum_l_o = tmp / sum_lo;
+				if (cum_l_o < thresh) {
+					index_A_occ.push_back(i);
+					outfile->Printf("\n Occupied orbital %d is partitioned to A with cumulative eigenvalue %8.8f",
+						i, cum_l_o);
+				}
+				else {
+					index_B_occ.push_back(i);
+				}
+			}
+			tmp = 0.0;
+			double cum_l_v = 0.0;
+			for (int i = 0; i < nrvirpi[0]; i++) {
+				tmp += lv->get(0, i);
+				cum_l_v = tmp / sum_lv;
+				if (cum_l_v < thresh) {
+					index_A_vir.push_back(i + offset_vec);
+					outfile->Printf("\n Virtual orbital %d is partitioned to A with cumulative eigenvalue %8.8f",
+						i + offset_vec, cum_l_v);
+				}
+				else {
+					index_B_vir.push_back(i + offset_vec);
+				}
+			}
+		}
+
+		int num_Ao = index_A_occ.size();
+		int num_Bo = index_B_occ.size();
+		int num_Av = index_A_vir.size();
+		int num_Bv = index_B_vir.size();
+
+		outfile->Printf("\n Build and semi-canonocalize new orbitals ------ \n");
+		SharedMatrix Ca_tilde(ref_wfn->Ca()->clone());
+
+		// Build and semi-canonicalize BO, AO, AV and BV blocks from rotated Ca()
+		auto C_bo = semicanonicalize_block(ref_wfn, Ca_tilde, index_B_occ, 0, true);
+		auto C_ao = semicanonicalize_block(ref_wfn, Ca_tilde, index_A_occ, 0, false);
+		auto C_av = semicanonicalize_block(ref_wfn, Ca_tilde, index_A_vir, 0, false);
+		auto C_bv = semicanonicalize_block(ref_wfn, Ca_tilde, index_B_vir, 0, true);
+
+		// Build the active block from original Ca_save
+		SharedMatrix C_A(new Matrix("Active_coeff_block", nirrep, nmopi, actv_a));
+		if (options.get_str("REFERENCE") == "CASSCF") {
+			if (options.get_bool("SEMICANON") == true) {
+				// Read active orbitals from original Ca and semi-canonicalize 
+				C_A->copy(semicanonicalize_block(ref_wfn, Ca_save, index_actv, 0, false));
+			}
+			else {
+				// Read active orbitals from original Ca and do not semi-canonicalize
+				for (int i = 0; i < actv_a[0]; ++i) {
+					C_A->copy(semicanonicalize_block(ref_wfn, Ca_save, index_actv, 0, true));
+				}
+			}
+		}
+
+		// Form new C matrix
+		SharedMatrix Ca_Rt(new Matrix("Ca rotated tilde", nirrep, nmopi, nmopi));
+
+		int offset = 0;
+		for (auto& C_block : { C_bo, C_ao, C_A, C_av, C_bv }) {
+			int nmo_block = C_block->ncol();
+			for (int i = 0; i < nmo_block; ++i) {
+				for (int mu = 0; mu < nmopi[0]; ++mu) {
+					double value = C_block->get(mu, i);
+					Ca_Rt->set(mu, offset, value);
+				}
+				offset += 1;
+			}
+		}
+
+		// Transform Fock matrix
+		ref_wfn->Fa()->transform(Ca_Rt);
+
+		// Update both the alpha and beta orbitals
+		ref_wfn->Ca()->copy(Ca_Rt);
+		ref_wfn->Cb()->copy(Ca_Rt);
```

Comment:
I was wondering if Cb is not necessary. Cb is Ca for RHF/ROHF/CASSCF.

Reply:
I would still update both MO sets. Just in case they are misused later.